File: circle_base_library/src/main/java/razerdp/github/com/common/mvp/models/entity/CommentInfo.java
package razerdp.github.com.common.mvp.models.entity;

import android.text.TextUtils;

import cn.bmob.v3.BmobObject;
import razerdp.github.com.common.manager.LocalHostManager;
import razerdp.github.com.common.mvp.models.IComment;

/**
 * Created by 大灯泡 on 2016/10/27.
 * <p>
 * 评论
 */

public class CommentInfo extends BmobObject implements icom{

    public interface CommentFields {
        String REPLY_USER = "reply";
        String MOMENT = "moment";
        String CONTENT = "content";
        String AUTHOR_USER = "author";
    }


    private MomentsInfo moment;
    private String content;
    private UserInfo author;
    private UserInfo reply;

    public MomentsInfo getMoment() {
        return moment;
    }

    public void setMoment(MomentsInfo moment) {
        this.moment = moment;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getCommentid() {
        return getObjectId();
    }

    public UserInfo getAuthor() {
        return author;
    }

    public void setAuthor(UserInfo author) {
        this.author = author;
    }

    public UserInfo getReply() {
        return reply;
    }

    public void setReply(UserInfo reply) {
        this.reply = reply;
    }

    public boolean canDelete() {
        return author != null && TextUtils.equals(author.getUserid(), LocalHostManager.INSTANCE.getUserid());
    }


    @Override
    public String toString() {
        return "CommentInfo{" +
                "moment=" + moment +
                ", content='" + content + '\'' +
                ", author=" + author.toString() + '\n' +
                ", reply=" + (reply == null ? "null" : reply.toString()) + '\n' +
                '}';
    }
}

File: baselib/src/main/java/razerdp/github/com/common/mvp/models/entity/CommentInfo.kt
package razerdp.github.com.common.mvp.models.entity

import android.text.TextUtils

import cn.bmob.v3.BmobObject
import razerdp.github.com.common.manager.LocalHostManager

/**
 * Created by 大灯泡 on 2016/10/27.
 *
 *
 * 评论
 */

 class CommentInfo : BmobObject() {


    var moment: MomentsInfo? = null
    var content: String? = null
    var author: UserInfo? = null
    var reply: UserInfo? = null

    val commentid: String
        get() = objectId

    interface CommentFields {
        companion object {
            val REPLY_USER = "reply"
            val MOMENT = "moment"
            val CONTENT = "content"
            val AUTHOR_USER = "author"
        }
    }

    fun canDelete(): Boolean {
        return author != null && TextUtils.equals(author!!.userid, LocalHostManager.INSTANCE.userid)
    }


    override fun toString(): String {
        return "CommentInfo{" +
                "moment=" + moment +
                ", content='" + content + '\'' +
                ", author=" + author!!.toString() + '\n' +
                ", reply=" + (if (reply == null) "null" else reply!!.toString()) + '\n' +
                '}'
    }
}
